src/test/scala/nl.knaw.dans.easy.bagstore/server/StoresServletSpec.scala

```diff
@@ -518,6 +520,41 @@ class StoresServletSpec extends TestSupportFixture
     }
   }
 
+  it should "fail when a an empty refbags.txt is provided" in {
+    createZipWithInvalidOrEmptyRefBag("")
+    val uuid = "11111111-1111-1111-1111-111111111111"
+    val bagId = BagId(UUID.fromString(uuid))
+    put(s"/store1/bags/$uuid", body = Files.readAllBytes(testBagUnprunedEmptyRefBag), basicAuthentication) {
+      status shouldBe 400
+      body should include(InvalidBagException(bagId, "the bag contains an empty refbags.txt").getMessage)
+    }
+  }
+
+  it should "fail when an invalid refbags.txt is provided" in {
+    val content = "invalid content"
+    createZipWithInvalidOrEmptyRefBag(content)
+    val uuid = "11111111-1121-1111-1111-111111111111"
+    val bagId = BagId(UUID.fromString(uuid))
+    put(s"/store1/bags/$uuid", body = Files.readAllBytes(testBagUnprunedEmptyRefBag), basicAuthentication) {
+      status shouldBe 400
+      body should include(s"Invalid UUID string: $content")
+    }
+  }
+
+  private def createZipWithInvalidOrEmptyRefBag(content: String) = {
+    Files.createFile(testBagsUnpruned.resolve("a").resolve("refbags.txt"))
+    val pw = new PrintWriter(testBagsUnpruned.resolve("a").resolve("refbags.txt").toFile)
+    try {
+      pw.write(content)
```

Comment:
I'm afraid using BetterFiles in this project needs another pull request, or even multiple ones.

Reply:
to my knowledge `better-files` is only used as a test dependency for now in `easy-bag-store`, right?

Reply:
Yes it is only used in test code as far as i know